### PR TITLE
Update documentation around public signing

### PR DESCRIPTION
**Pull request by jaredpar**
The feature know as "Open Source Signing" was implemented in the compilers as "Public Signing".  The term OSS was dropped after conversations with Immo as it was applicable to other projects.  It was simply signing with the public key hence better named "Public Signing".  Missed updating this documentation when that change was made.

### CODE REVIEW

**Comment by jaredpar** [comment]
CC @terrajobst

File: Documentation/README.md [hunk 1]
```diff
@@ -29,7 +29,7 @@ Project Docs
 - [Branching Guide](project-docs/branching-guide.md)
 - [API Review Process](project-docs/api-review-process.md)
 - [Strong Name Signing](project-docs/strong-name-signing.md)
-- [Open Source Signing](project-docs/oss-signing.md)
+- [Public Source Signing](project-docs/public-signing.md)
```
Comment (mellinoe) [inline Documentation/README.md]: Should this be "Public Signing" ?

File: Documentation/project-docs/oss-signing.md [hunk 2]
```diff
@@ -1,18 +0,0 @@
-Open Source Signing
```
Comment (akoeplinger) [inline Documentation/project-docs/oss-signing.md]: I think it'd make sense to keep this document for a while to not break links and explain that OSS Signing was the old term and link to the public signing doc.

File: Documentation/README.md (hunk 1 above)
Comment (jaredpar) [inline Documentation/README.md]: Yep.  I will fix.

File: Documentation/project-docs/oss-signing.md (hunk 2 above)
Comment (jaredpar) [inline Documentation/project-docs/oss-signing.md]: Good point.  I'll re-add this and point it to the new document.

**Comment by jaredpar** [comment]
Responded to PR feedback.

**Pull request merged**